smoke-test/src/test/java/org/opennms/smoketest/rest/FileEditorIT.java

```diff
@@ -0,0 +1,364 @@
+
+            Response resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + file.getName(), "POST", body);
+            resp.close();
+
+            body = new MultipartBody.Builder().setType(MultipartBody.FORM)
+                    .addFormDataPart("upload", "OpenNMS Smoke Test")
+                    .build();
+
+            resp = postRequest(STACK.opennms().getBaseUrlExternal() + REST_FILESYSTEM + "/contents?f=" + file.getName(), "POST", body);
+
+            Assert.assertEquals(400, resp.code());
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        }
+    }
+
+    /**
+     * Create a Basic Header for the Authentication to server
+     *
+     * @return
+     */
+    private String authHeader(String username, String password) {
+        String auth = username + ":" + password;
+        byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(Charset.forName("US-ASCII")));
+        return "Basic " + new String(encodedAuth);
+    }
+
+    /**
+     * Generic method to start a request
+     *
+     * @param url
+     * @param method
+     * @param body
+     * @return
+     * @throws IOException
+     */
+    private Response postRequest(String url, String method, RequestBody body) throws IOException {
+        LOG.info("creating request");
+
+        OkHttpClient client = new OkHttpClient().newBuilder()
+                .build();
+
+        Request request = new Request.Builder()
+                .url(url)
+                .method(method, body)
+                .addHeader("Authorization", authHeader(USERNAME, PASSWORD))
+                .build();
+        Response response = client.newCall(request).execute();
+
+        LOG.info(response.toString());
+        return response;
+    }
+
+    /**
+     * Add an user over API
+     */
+    private void addUserAPI() {
+        LOG.info("User creation request");
+        final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        JAXB.marshal(createUser("editor", "File Editor", "editor@opennms.org", "21232F297A57A5A743894A0E4A801FC3" /* admin */, "ROLE_FILESYSTEM_EDITOR", "ROLE_USER"), outputStream);
+
+        final HttpPost post = new HttpPost(STACK.opennms().getBaseUrlExternal().toString() + "/opennms" + "/rest/users");
+        post.setEntity(new StringEntity(new String(outputStream.toByteArray()), ContentType.APPLICATION_XML));
+        Integer response = 0;
+        try {
+            response = doRequest(post);
+        } catch (IOException e) {
+            LOG.error(e.toString());
+        } catch (InterruptedException e) {
+            LOG.error(e.toString());
+        }
+
+        LOG.info(response.toString());
+    }
+
+    /**
+     * API request to create a user
+     * @param request
+     * @return
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    protected Integer doRequest(final HttpRequestBase request) throws IOException, InterruptedException {
```

Comment:
Now that you got that working without `AbstractOpenNMSSeleniumHelper`. Can you try to reuse those methods from `AbstractOpenNMSSeleniumHelper` by making them public static or by Abstracting them in another class